src/Domain/RegularGivingService.php

```diff
@@ -2,15 +2,160 @@
+     * @throws \Doctrine\DBAL\Exception\ServerException
+     * @throws \Doctrine\ORM\Exception\ORMException
+     * @throws \MatchBot\Client\CampaignNotReady
+     * @throws \MatchBot\Client\NotFoundException
+     * @throws \Symfony\Component\Notifier\Exception\TransportExceptionInterface
+     */
+    public function setupNewMandate(
+        PersonId $donorID,
+        Money $amount,
+        Campaign $campaign,
+        bool $giftAid,
+        DayOfMonth $dayOfMonth,
+    ): RegularGivingMandate {
+        if (! $campaign->isRegularGiving()) {
+            throw new WrongCampaignType(
+                "Campaign {$campaign->getSalesforceId()} does not accept regular giving"
+            );
+        }
+
+        $charityId = Salesforce18Id::ofCharity(
+            $campaign->getCharity()->getSalesforceId() ?? throw new \Exception('missing charity SF ID')
+        );
+
+        /**
+         * For now we assume this exists - @todo-regular-giving ensure that for all accounts (or all accounts that
+         * might need it) the account is in the DB with the UUID filled in before this point.
+         */
+        $donor = $this->donorAccountRepository->findByPersonId($donorID);
+        if ($donor === null) {
+            throw new \Exception("donor not found with ID {$donorID->id}");
+        }
+
+        $mandate = new RegularGivingMandate(
+            donorId: $donorID,
+            amount: $amount,
+            campaignId: Salesforce18Id::ofCampaign(
+                $campaign->getSalesforceId() ?? throw new \Exception('missing campaign SF ID')
+            ),
+            charityId: $charityId,
+            giftAid: $giftAid,
+            dayOfMonth: $dayOfMonth,
+        );
+
+        $this->entityManager->persist($mandate);
+
+        $firstDonation = new Donation(
+            amount: $amount->toNumericString(),
+            currencyCode: $amount->currency->isoCode(),
+            paymentMethodType: PaymentMethodType::Card,
+            campaign: $campaign,
+            charityComms: false,
+            championComms: false,
+            pspCustomerId: $donor->stripeCustomerId->stripeCustomerId,
+            optInTbgEmail: false,
+            donorName: $donor->donorName,
+            emailAddress: $donor->emailAddress,
+            countryCode: $donor->getBillingCountryCode(),
+            tipAmount: '0',
+            mandate: $mandate,
+            mandateSequenceNumber: DonationSequenceNumber::of(1)
+        );
+        $this->donationService->enrollNewDonation($firstDonation);
+        if (! $firstDonation->isFullyMatched()) {
+            throw new NotFullyMatched(
+                "Donation could not be fully matched, need to match {$firstDonation->getAmount()}," .
+                " only matched {$firstDonation->getFundingWithdrawalTotal()}"
+            );
+        }
+        // @todo-regular-giving - throw if first donation is not fully matched unless donor has said they're OK with
+        //                        that.
```

Comment:
```suggestion
```

Reply:
And maybe the todo's below? I've slightly lost the thread of which parts were already on `develop`, could be the later parts were done already?